src/platform/orchestrator/src/MpiApi.cpp

```diff
@@ -0,0 +1,489 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "CommonUtils.h"
+#include "MpiApi.h"
+#include "Http.h"
+#include "ModulesManager.h"
+
+#include <sstream>
+#include <iostream>
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+#include <rapidjson/stringbuffer.h>
+
+static const char* g_clientName = "ClientName";
+static const char* g_maxPayloadSizeBytes = "MaxPayloadSizeBytes";
+static const char* g_clientSession = "ClientSession";
+static const char* g_componentName = "ComponentName";
+static const char* g_objectName = "ObjectName";
+static const char* g_payload = "Payload";
+
+static const char* g_contentLength = "Content-Length";
+static const char* g_contentType = "Content-Type";
+static const char* g_contentTypeJson = "application/json";
+
+static const char* g_socketPrefix = "/run/osconfig";
+static const char* g_mpiSocket = "/run/osconfig/mpid.sock";
+
+// TODO: dynamic size for reading
+static const size_t g_maxPayloadSize = 4096;
+
+// TEMPORARY: global map of all the sessions (will be maintained by orchestrator)
+static std::map<std::string, MPI_HANDLE> g_sessions;
+
+static Router router;
+static Server server;
+
+std::string GenerateRandomString(int length)
+{
+    std::string s;
+    s.resize(length);
+    for (int i = 0; i < length; i++)
+    {
+        s[i] = 'a' + (rand() % 26);
+    }
```

Comment:
Later (not now) change to GUID-like hex sequence. Random chars can result into some unintended words.